libraries/api_helper.rb

```diff
@@ -29,7 +29,7 @@ def login(host, port, user, password)
       )
 
       # sorry for the fancy hackery - rubists are welcome to make this better
-      response['set-cookie'][/grafana_sess=(\w+);/, 1]
+      response['set-cookie'][/#{GrafanaCookbook::CookieHelper.cookie_name}=(\w+);/, 1]
```

Comment:
We need to somehow abstract this `cookie_name` so that a user can input it and it is passed into these methods, else if someone changes the default cookie name these will always fail :/ 

Do you have any ideas on how to achieve this?

Reply:
@Xorima If the user calls `GrafanaCookbook::CookieHelper.cookie_name = 'new_cookie_name'` , it will set it to that cookie name across all the resources and inside the library call.  I put it into the README.md.  It gives the user control, but not sure this is the best solution.  At one point I had it part of the config_session resource, but realized it depended the order the user specified the resources.  Since there isn't a recipe in this cookbook, it is hard to manage :/ global settings like this.

Reply:
you could stick it in the `run_state` cache <https://docs.chef.io/recipes.html#node-run-state>, and have your library check if it exists in there it would return it from there

If they choose not to use the resource to configure grafana that would still cause them issues but I am of the opinion cater for the 95% use-case (chef the install as well as the api calls)

Reply:
I did initially look at using the run_state, but from the research I did , it didn't appear possible to access the run_state from inside libraries.  I can try and see if I can find something with more research.